test-data/unit/check-typevar-values.test

```diff
@@ -631,3 +631,18 @@ def g(s: S) -> Callable[[S], None]: ...
 def f(x: S) -> None:
     h = g(x)
     h(x)
+
+[case testTypeVarWithTypedDictBoundInIndexExpression]
+from typing import TypeVar
+from typing_extensions import TypedDict
+
+class Data(TypedDict):
+    x: int
+
+
+T = TypeVar("T", bound=Data)
```

Comment:
Will the same work for `bound=Union[Data, Other]`?

Reply:
yes, it works because `Union` is handled in `visit_index_with_type` by processing Union items recursively. Added test for this case.